File: java/LeapYear.java
import java.util.Scanner;

public class LeapYear {
    public static void main(String[] args) {

        System.out.println("Enter a year to check if it is a leap year or not");
        int year = new Scanner(System.in).nextInt();

        if ((year % 4 == 0 && year % 100 != 0) || (year % 400 == 0)) {
            System.out.println(year + " is leap year ");
        } else {
            System.out.println(year + " is not leap year ");
        }
    }
}

File: js/leapyear.js
// program to check leap year

function checkLeapYear(year) {
    // checking for the number to be leap year
    if((0 == year % 4) && (0 != year % 100) || (0 == year % 400)) {
        console.log(year + ' is a leap year');
    } else {
        console.log(year + ' is not a leap year');
    }
}

// take input
const year = prompt('Enter a year to check:');
checkLeapYear(year);
